pysatModels/tests/test_models.py

```diff
@@ -1,38 +1,43 @@
 import pytest
 
 import pysatModels
-from pysat.tests.instrument_test_class import generate_instrument_list
+from pysat.utils import generate_instrument_list
 from pysat.tests.instrument_test_class import InstTestClass
 
-instruments = generate_instrument_list(package=pysatModels.models)
+instruments = generate_instrument_list(inst_loc=pysatModels.models)
 
 method_list = [func for func in dir(InstTestClass)
                if callable(getattr(InstTestClass, func))]
+
 # Search tests for iteration via pytestmark, update instrument list
 for method in method_list:
     if hasattr(getattr(InstTestClass, method), 'pytestmark'):
         # Get list of names of pytestmarks
-        Nargs = len(getattr(InstTestClass, method).pytestmark)
-        names = [getattr(InstTestClass, method).pytestmark[j].name
-                 for j in range(0, Nargs)]
+        mark_name = [mod_mark.name for mod_mark
+                     in getattr(InstTestClass, method).pytestmark]
+
         # Add instruments from your library
-        if 'all_inst' in names:
-            mark = pytest.mark.parametrize("name", instruments['names'])
+        if 'all_inst' in mark_name:
+            mark = pytest.mark.parametrize("inst_name", instruments['names'])
             getattr(InstTestClass, method).pytestmark.append(mark)
-        elif 'download' in names:
-            mark = pytest.mark.parametrize("inst", instruments['download'])
+        elif 'download' in mark_name:
+            mark = pytest.mark.parametrize("inst_dict",
+                                           instruments['download'])
             getattr(InstTestClass, method).pytestmark.append(mark)
-        elif 'no_download' in names:
-            mark = pytest.mark.parametrize("inst", instruments['no_download'])
+        elif 'no_download' in mark_name:
+            mark = pytest.mark.parametrize("inst_dict",
+                                           instruments['no_download'])
             getattr(InstTestClass, method).pytestmark.append(mark)
 
 
 class TestModels(InstTestClass):
 
     def setup(self):
-        """Runs before every method to create a clean testing setup."""
-        self.package = pysatModels.models
+        """Runs before every method to create a clean testing setup
+        """
+        self.inst_loc = pysatModels.models
 
     def teardown(self):
-        """Runs after every method to clean up previous testing."""
-        del self.package
+        """Runs after every method to clean up previous testing
+        """
+        del self.inst_loc
```

Comment:
You should use the class level setup and teardown as well.  We've only got one file that currently uses download support, but this will ensure that the main directory is not messed with by the tests.  See example at:

https://github.com/pysat/pysatNASA/blob/14d77d4104a71f750ce20c785b7838adc6294427/pysatNASA/tests/test_instruments.py#L42-L66

Reply:
So, not sure how this example is supposed to be applied.  Also confused as to why custom setup/teardown methods were created.

Reply:
Ideally, we should download to a temporary directory during tests to both simulate a new install and protect the developer's data directory (test files are removed at the end of the tests).  This is a feature that was lost early in the development of the tests, which is why it's missing here.  Since the load tests need to use the same directory as download, a class-level setup is required (runs once at the beginning of tests).  

Should just be copy and paste with the new `inst_loc` applied, and importing `tempfile` at the top.

Reply:
Maybe it'd be best to talk about this at the meeting tomorrow.